### PR TITLE
AQL timezone extension

### PR DESCRIPTION
**Pull request by Simran-B**
### Description

Add timezone parameter to functions, add examples, formatting and style

Port of https://github.com/arangodb/docs/pull/1290

- https://github.com/arangodb/arangodb/pull/18074
- https://github.com/arangodb/arangodb/pull/20339

#### Upstream PRs

- 3.10:
- 3.11:
- 3.12: arangodb/enterprise-preview:devel-nightly

### CODE REVIEW

File: site/content/3.12/aql/functions/date.md [hunk 1]
```diff
@@ -401,34 +528,68 @@ description: ''
 RETURN DATE_MILLISECOND("2020-08-29T16:30:05.123")
 ```
 
+```aql
+---
+name: datesec2
```
Comment (Simran-B) [inline site/content/3.12/aql/functions/date.md:533]: The duplicate example name resulted in this example overwriting the previous one for DATE_SECOND(), apparently without a warning/error!
```suggestion
name: datemilsec2
```

Comment (nerpaula) [reply site/content/3.12/aql/functions/date.md:533]: Hmm. Let's update its name and re-run the example generation. I will create a ticket to fix this.

**Review by nerpaula — APPROVED** [review]

**Pull request merged**